plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/NewPackageAction.java

```diff
@@ -41,13 +42,37 @@
  */
 @Singleton
 public class NewPackageAction extends AbstractNewResourceAction {
+
+    private final AppContext appContext;
     private final InputValidator nameValidator = new NameValidator();
 
     @Inject
-    public NewPackageAction(JavaResources javaResources, JavaLocalizationConstant localizationConstant) {
+    public NewPackageAction(JavaResources javaResources,
+                            JavaLocalizationConstant localizationConstant,
+                            AppContext appContex    ) {
         super(localizationConstant.actionNewPackageTitle(),
               localizationConstant.actionNewPackageDescription(),
               javaResources.packageIcon());
+        this.appContext = appContex;
+    }
+
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent e) {
+        CurrentProject project = appContext.getCurrentProject();
```

Comment:
`Project Action` will extends `Abstract Perspective Action`.
`New Java Source File Action` will extends `Abstract New Resource Action`.
`New Package Action` will extends `Abstract New Resource Action` which will extends `Project Action`

Yes, we need to refactor this, but I think that this is for another issue...